ownCloud/Client/Actions/Actions+Extensions/DeleteAction.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  DeleteAction.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 12/11/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import ownCloudSDK
+
+class DeleteAction : Action {
+	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.delete") }
+	override class var category : ActionCategory? { return .destructive }
+	override class var name : String? { return "Delete".localized }
+
+	// MARK: - Extension matching
+	override class func applicablePosition(forContext: ActionContext) -> ActionPosition {
+		// Examine items in context
+		return .last
+	}
+
+	// MARK: - Action implementation
+	override func run() {
+		guard context.items.count > 0, let viewController = context.viewController else {
+			completionHandler?(NSError(ocError: .errorInsufficientParameters))
+			return
+		}
+
+		beforeRunHandler?()
+
+		let item = context.items[0]
```

Comment:
To futureproof the actions, it should operate on all passed items (same for all other actions).